### PR TITLE
Rewrite XCFramework slice selection using plist metadata

### PR DESCRIPTION
**Pull request by igor-makarov**
This PR modifies the XCFramework selection script in such a way that it uses the actual Info.plist values of the XCF to determine slice eligibility.

I've seen a while ago that the script was using directory names for differentiation and I had a feeling that this is not entirely correct.

I've added 2 new examples to the build and verified that they are testing the right stuff:
* `Excluded-Archs-Example` - tests the common exclusion method of `EXCLUDED_ARCHS` when a company doesn't want to make an XCFramework for all the platforms.
* `Oddly-Named-XCF-Slices-Example` - has slices with odd names. This would not have worked in the previous script.

note: integration spec PR https://github.com/CocoaPods/cocoapods-integration-specs/pull/332

### CODE REVIEW

**Comment by igor-makarov** [comment]
@amorde can you tell me if you have any concrete examples of oddly named slices?

**Comment by igor-makarov** [comment]
going to remove the change to error

File: lib/cocoapods/generator/copy_xcframework_script.rb [hunk 1]
```diff
@@ -93,29 +99,15 @@ def script
...
-      # Let's say the folder name is: ios-armv7_armv7s_arm64_arm64e/CoconutLib.framework
-      # We match the following: -armv7_, _armv7s_, _arm64_ and _arm64e/.
-      # If we have a specific variant: ios-i386_x86_64-simulator/CoconutLib.framework
-      # We match the following: -i386_ and _x86_64-
-      # When the .xcframework wraps a static library, the folder name does not include
-      # any .framework. In that case, the folder name can be: ios-arm64_armv7
-      # We also match _armv7$ to handle that case.
-      local target_arch_regex="[_\\-]${target_arch}([\\/_\\-]|$)"
-      if ! [[ "${paths[$i]}" =~ $target_arch_regex ]]; then
+      if ! echo "${slice_archs}" | tr " " "\\n" | grep -F -q -x "$target_arch"; then
```
Comment (amorde) [inline lib/cocoapods/generator/copy_xcframework_script.rb:110]: do you have some context on this change? it's unclear to me what this is doing. is it splitting the archs into separate lines and then using `grep` to see if any lines match?

Comment (igor-makarov) [reply lib/cocoapods/generator/copy_xcframework_script.rb:110]: @amorde exactly - it's an internet-suggested way to check that `$target_arch` is in the `$slice_archs` list (which is space-separated).

**Review by amorde — APPROVED** [review]

**Review by dnkoutso — APPROVED** [review]

**Comment by igor-makarov** [comment]
Merged https://github.com/CocoaPods/cocoapods-integration-specs/pull/332 - now merging this.

**Pull request merged**